ebos/eclgenericproblem_impl.hh

```diff
@@ -169,6 +169,22 @@ readRockParameters_(const std::vector<Scalar>& cellCenterDepths)
         const auto& num = eclState_.fieldProps().get_int(rock_config.rocknum_property());
         for (size_t elemIdx = 0; elemIdx < numElem; ++ elemIdx) {
             rockTableIdx_[elemIdx] = num[elemIdx] - 1;
+            if (!rockCompPoroMult_.empty() &&
+                rockTableIdx_[elemIdx] >= rockCompPoroMult_.size()) {
+                throw std::runtime_error("Rock compaction table index for elem " +
+                                         std::to_string(elemIdx) +
+                                         " read from " + rock_config.rocknum_property() +
+                                         " is larger than number of tables " +
```

Comment:
I agree, this is the common case.  There is another one though and we should report that one too.  If the model erroneously specifies a `*NUM` value less than 1 in an active cell, then we're equally in trouble.  On another note, if this function (`readRockParameters_()`) runs on all MPI ranks, then the `elemIdx` is local to each rank and **probably** won't make much sense to the user.  How much work would it be to translate back to a global (I,J,K) tuple?

In any case I think we should report a message of the form

> Rock compaction table "num[elemIdx]" ... is out of bounds

to encompass the `NUM < 1` possibility too.

Reply:
done and done.